docs/UsingOurScripts.md

```diff
@@ -0,0 +1,121 @@
+    "configDirectories"      : {
+        // Build configs directory containing node details, os, arch, testlists, etc
+        "build"              : "pipelines/jobs/configurations",
+        // Nightly configs directory containing execution frequency, weekly tags, platforms to build.
+        "nightly"            : "pipelines/jobs/configurations",
+        // Bash platform script directory containing jdk downloading and toolchain setups.
+        "platform"           : "build-farm/platform-specific-configurations"
+    },
+    // Job script paths (relative to this repository root)
+    "scriptDirectories"      : {
+        // Upstream scripts directory containing the 1st files that are executed by the openjdkx-pipeline jobs.
+        "upstream"           : "pipelines/build",
+        // Upstream script file containing the 1st script that is executed by the weekly-openjdk8-pipeline jobs.
+        "weekly"             : "pipelines/build/common/weekly_release_pipeline.groovy",
+        // Downstream script file containing the 1st script that is executed by the jdkx-platform-arch-variant jobs.
+        "downstream"         : "pipelines/build/common/kick_off_build.groovy",
+        // Base script file containing the 2nd script that is executed by the pipeline_jobs_generator_jdkxx jobs
+        "regeneration"       : "pipelines/build/common/config_regeneration.groovy",
+        // Base PR tester file script file containing the 2nd script that is executed by the pipeline_jobs_generator_jdkxx jobs
+        "tester"             : "pipelines/build/prTester/pr_test_pipeline.groovy"
+    },
+    // Job base file (the main file which is called after the 1st setup script file) paths (relative to this repository root)
+    "baseFileDirectories": {
+        // Upstream pipeline file script containing the 2nd script that is executed by the openjdkx-pipeline jobs
+        "upstream"           : "pipelines/build/common/build_base_file.groovy",
+        // Upstream pipeline file script containing the 2nd script that is executed by the jdkx-platform-arch-variant jobs
+        "downstream"         : "pipelines/build/common/openjdk_build_pipeline.groovy"
+    },
+    // Script to import the adopt groovy class library (relative to this repository root)
+    "importLibraryScript"    : "pipelines/build/common/import_lib.groovy"
+}
+```
+
+### How do I know which parameter the jenkins job will use?
+
+The scripts have been designed with a set hierarchy in mind when choosing which parameter to use:
+
+```md
+1. JENKINS PARAMETERS (highest priority, args entered here will be what the build scripts use over everything else)
+2. USER JSON (medium priority, args entered here will be used when a jenkins parameter isn't entered)
+3. ADOPT JSON (final priority, when jenkins parameters AND a user json arg can't be validated, the script will checkout to this repository and use our defaults json (linked above))
```

Comment:
Is this the right order? I'd have thought that the user over ride would be the last one.

Reply:
User override should ideally be the first otherwise the scripts would just pull in Adopt's data no questions asked. Working vice versa would defeat the point of the PR and Adam's review above.

If a user doesn't want to use Adopt's data and wants to use their own, they should follow the instructions in the guide and either point their defaults.json at whatever the filepath to their data is and leave the Jenkins parameters alone OR stick the path in the Jenkins parameters and leave the defaults.json alone.

Vice versa, if a user wants to use Adopt's data and not have the jobs detect their own data (if they have it), they should remove or invalidate the corresponding entry in their defaults.json and the scripts will automatically pull and use Adopt's scripts.

Happy to discuss this further or validate any points that are confusing? I recognise it's a lot of changes and a 180 degree behaviour in terms of the pipelines 🙂

Reply:
Let's have a call - I get what the intention is here, it just goes slightly against the typical design for these sorts of systems where the defaults are laid ontop of each other (and are well documented and disseminated!) and the user overrides come last.

Reply:
The final decision has been to open up a follow up issue to address expanding the current `USE_ADOPT_SHELL_SCRIPTS` parameter to cover all instances of the checking out (see [checkling comment](https://github.com/AdoptOpenJDK/openjdk-build/pull/2132#issuecomment-735996265).

This will likely require refactoring and upgrading once the repo strip out goes through so we will need to keep that in mind after this is merged

Reply:
https://github.com/AdoptOpenJDK/openjdk-build/issues/2449